### PR TITLE
feat(bigquery): implement BigQuery  operations

### PR DESCRIPTION
**Pull request by krzysztof-kwitt**

### CODE REVIEW

File: ibis/backends/bigquery/registry.py [hunk 1]
```diff
@@ -490,6 +491,60 @@ def _array_repeat(t, op):
...
+        f"WHERE {' AND '.join(cond)}"
+        f")"
+    )
+
+
+def _capitalize(t, op):
+    return f"CONCAT(UPPER(SUBSTR({t.translate(op.arg)}, 1, 1)), SUBSTR({t.translate(op.arg)}, 2))"
+
+
+def _translate(t, op):
```
Comment (cpcloud) [inline ibis/backends/bigquery/registry.py:516]: I'm not sure we have sufficient test coverage for this function, so let's leave it out. This API is supposed to be modeled after Python's [`str.translate`](https://docs.python.org/3/library/stdtypes.html#str.translate) method, but to be honest I don't think we have a non-trivial test for this API.

What do you think about implementing this operation in a follow-up PR and leaving it out on this one for now?

File: ibis/backends/bigquery/registry.py [hunk 2]
```diff
@@ -490,6 +490,40 @@ def _array_repeat(t, op):
...
+    cond = [f"index >= {_neg_idx_to_pos(arg, t.translate(op.start))}"]
+    if op.stop:
+        cond.append(f"index < {_neg_idx_to_pos(arg, t.translate(op.stop))}")
+    return (
+        f"ARRAY("
+        f"SELECT el "
+        f"FROM UNNEST({arg}) AS el WITH OFFSET index "
+        f"WHERE {' AND '.join(cond)}"
+        f")"
+    )
```
Comment (cpcloud) [inline ibis/backends/bigquery/registry.py:508]: Whoa, gnarly implementation :)

**Review by cpcloud — APPROVED** [review]
LGTM, I'm going to run the BigQuery tests locally and then merge on green!

**Comment by cpcloud** [comment]
Fixed up an `XPASS`ing test `test_order_by_random`. Everything else is green:

```
❯ pytest -m bigquery -n auto --dist=loadgroup --snapshot-update --allow-snapshot-deletion
======================================================================================== test session starts =========================================================================================
platform linux -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0
benchmark: 4.0.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Using --randomly-seed=2197632894
rootdir: /home/cloud/src/ibis, configfile: pyproject.toml
plugins: mock-3.10.0, repeat-0.9.1, benchmark-4.0.0, xdist-3.1.0, hypothesis-6.61.0, snapshot-0.9.0, clarity-1.0.1, profiling-1.7.0, randomly-3.12.0, cov-4.0.0
gw0 [1005] / gw1 [1005] / gw2 [1005] / gw3 [1005] / gw4 [1005] / gw5 [1005] / gw6 ok / gw7 [1005] / gw8 [1005] / gw9 ok / gw10 [1005] / gw11 [1005] / gw12 [1005] / gw13 [1005] / gw14 ok / gw15 [1005gw0 [1005] / gw1 [1005] / gw2 [1005] / gw3 [1005] / gw4 [1005] / gw5 [1005] / gw6 [1005] / gw7 [1005] / gw8 [1005] / gw9 ok / gw10 [1005] / gw11 [1005] / gw12 [1005] / gw13 [1005] / gw14 ok / gw15 [gw0 [1005] / gw1 [1005] / gw2 [1005] / gw3 [1005] / gw4 [1005] / gw5 [1005] / gw6 [1005] / gw7 [1005] / gw8 [1005] / gw9 ok / gw10 [1005] / gw11 [1005] / gw12 [1005] / gw13 [1005] / gw14 [1005] / gwgw0 [1005] / gw1 [1005] / gw2 [1005] / gw3 [1005] / gw4 [1005] / gw5 [1005] / gw6 [1005] / gw7 [1005] / gw8 [1005] / gw9 [1005] / gw10 [1005] / gw11 [1005] / gw12 [1005] / gw13 [1005] / gw14 [1005] / gw15 [1005]
xx.......xx....x....xxx.....xx..xx.xx..x.xxx....x...x.x.x..xxx.......x...x.x....x.....xxxx.........xx..x...xxx.xxx.xx...x.......xxx.x..x....x.x.xx.x..x.x.xxx..x.xxx...xxx.x..xxx...x......... [ 18%]
.....x.........x........................x...........................................x...........x.x...xx......x.x.xxx..xxx...xx.xx.....x...........x....x....x.x.x...x.x..x.xxxxxx.xxxxx..x... [ 37%]
x.x.xxx..x..x.x.x....x..x.....x.....xx....x.xxx.x.....x..x..xx...........x......x..........x.....x.....xxx......x....x.x...xx.x....x.xx..xx.........x.x.....xx...xxxxx..x..x..xx.............. [ 56%]
x............xx.xx......F...x.x.x.x............x..................x..x.x....xx.x....x......x...xx.....x.x.....x....x..x.sx.xx...x.x..xx.........................x..................xx.x......x [ 75%]
x...........x..x..xx..x.x....x.x..............xx.x............................................................................................................................................ [ 94%]
...................x..........s...x..x.....x...........                                                                                                                                        [100%]
============================================================================================== FAILURES ==============================================================================================
___________________________________________________________________________________ test_order_by_random[bigquery] ___________________________________________________________________________________
[gw13] linux -- Python 3.10.9 /nix/store/rkjc53gdm707c5bk9n6afrk16zv1i49j-python3-3.10.9-env/bin/python3.10
[XPASS(strict)] Feature not yet exposed in bigquery
====================================================================================== short test summary info =======================================================================================
FAILED ibis/backends/tests/test_generic.py::test_order_by_random[bigquery]
================================================================= 1 failed, 782 passed, 2 skipped, 220 xfailed in 106.67s (0:01:46) ==================================================================
```

File: ibis/backends/bigquery/registry.py (hunk 1 above)
Comment (krzysztof-kwitt) [reply ibis/backends/bigquery/registry.py]: Done

**Comment by cpcloud** [comment]
Great work, thank you!

**Pull request merged**